net/socks5/socks5.go

```diff
@@ -143,7 +150,8 @@ type Conn struct {
 	clientConn net.Conn
 	request    *request
 
-	udpClientAddr net.Addr
+	udpClientAddr  net.Addr
+	udpTargetConns syncs.Map[string, net.Conn]
```

Comment:
this seems pretty expensive (both the string allocation and the mutex) per packet.

at least key it off netip.AddrPort instead of string?

But ideally we wouldn't even need a mutex per packet.

Reply:
This Conn cache may be accessed by multiple go routines at the same time so I feel that the locking is unavoidable🤔 But for applications that don't send UDP packets to multiple different addresses, the cost will not be too great? Of course netip.AddrPort would be a bit better

BTW I have a suggestion that maybe something like LRU or Timeout is needed to be applied to the cache, to avoid  running out of port numbers when some program send packets to too many destinations. Maybe `maxInFlightConnectionAttemptsPerClient` in `netstack.go` will be a suitable limitation to the size of cache.

Reply:
We may not need locking at all. The udpTargetConns is only used within the getOrDialTargetConn function, which is exclusively called by the goroutine handling client-to-target packet forwarding.

Given this single-threaded access pattern, we can simplify our approach by using a regular `map[netip.AddrPort]net.Conn` without any synchronization overhead. This would address the concerns about per-packet mutex operations and string allocations.